### ISSUE TITLE
Pay attention to the new specs

### ISSUE THREAD

**Issue by ghost** [llvm-dcpu16/llvm-dcpu16#143]
http://www.reddit.com/r/dcpu16/comments/sqfre/rfe_dcpu16_11/

**Comment by ghost** [comment]
I think it boils down to this TODO list:
- Change the stack pointer from I to SP
- Rename register O to EX
- Add signed MUL/DIV
- Add ASR
- Support all new IF opcodes

**Comment by krasin** [comment]
- Change the Notch order to the normal one :)

**Comment by ghost** [comment]
_mhh_ I'm not too sure about the order thing ... since he only said things like "changed the naming" and "evalutation of a,b". But if he really changed the order, then it's not too hard for us, since we have a pretty slick TableGen right now (and the special lowering functions of course).

**Comment by krasin** [comment]
`1 | 0x01 | SET b, a | sets b to a`

http://dcpu.com/highnerd/dcpu16.txt

**Comment by ghost** [comment]
Uhh didn't saw that one!

**Comment by Blei** [comment]
Oh noez, now when I don't have time he updates the spec D: Maybe on the weekend...

**Comment by ghost** [comment]
I will update the small parts (like renaming regsters, adding new inst and switching the operands) this evening, but I won't implement the IF codes, since I don't feel like I know exactly how they are working. (@Blei that might be your job then)

**Comment by ghost** [comment]
Okay 1.3 is out:
http://dcpu.com/highnerd/dcpu16_1_3.txt
- Fixed a few typos
- Added MVI. Unfortunately, it makes assumptions about two registers, but hopefully it's simple and usable.
- Added ADX and SUX, add and subtract with carry
- Clarified the interrupt behavior
- Removed the built in 60 hz clock, it will be an external device

**Comment by ghost** [comment]
Okay, shoot me but the order is still the same!

Old:

```
0x1: SET a, b - sets a to b
```

New:

```
 1 | 0x01 | SET b, a | sets b to a
```

Just the naming changed.

**Comment by ghost** [comment]
Question: Shouldn't we replace the unsigned MUL with the signed MLI? As far as I can see, LLVM only cares for signed multiplication.

**Comment by Blei** [comment]
If we ignore the overflow part in EX, signed and unsigned multiplication is the same. There are some opcodes `SMUL_LOHI`, `UMUL_LOHI`, that care about those things. At least I think that's the way it works.

**Comment by ghost** [comment]
Okay and those are right now expanded. I will leave MUL as it is right now.

**Comment by ghost** [comment]
Partly fixed by 258221b195acbd1b4e8a28b15daaf5989e771bb7.

TODO:
- Support all new IF opcodes

**Comment by ghost** [comment]
Closed, because there is a new version at #147
